### PR TITLE
Add helper functions for Google.Protobuf modules

### PR DESCRIPTION
**Pull request by btkostner**
Replaces https://github.com/elixir-protobuf/google-protos/pull/21. Should be merged after https://github.com/elixir-protobuf/protobuf/pull/353.

### CODE REVIEW

**Review by Nezteb — APPROVED** [review]

**Review by ddengler — APPROVED** [review]

File: lib/google/protobuf.ex [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+  ## Examples
+
+      iex> to_time_unit(%Google.Protobuf.Duration{seconds: 10})
+      {10, :second}
+
+      iex> to_time_unit(%Google.Protobuf.Duration{seconds: 20, nanos: 100})
+      {20_000_000_100, :nanosecond}
+
+  """
+  @spec to_time_unit(Google.Protobuf.Duration.t()) :: {integer(), System.time_unit()}
```
Comment (btkostner) [inline lib/google/protobuf.ex:18]: Heads up, now that we have https://hexdocs.pm/elixir/main/Duration.html, we should change these functions to match.

Comment (v0idpwn) [reply lib/google/protobuf.ex:18]: We would need to bump elixir requirement, though

Comment (v0idpwn) [reply lib/google/protobuf.ex:18]: I think it's fine to keep as it is?

Comment (v0idpwn) [reply lib/google/protobuf.ex:18]: Maybe we can **accept** it in `from_time_unit` if using Elixir >= 1.17, but continue returning ` {integer(), System.time_unit()}` on `to_time_unit`

**Review by v0idpwn — APPROVED** [review]

File: lib/google/protobuf.ex (hunk 1 above)
Comment (whatyouhide) [reply lib/google/protobuf.ex:18]: If we don't return `Duration`, we're introducing a function now that is bound to be deprecated soon. I think I wouldn't define `to_time_unit/1` in the first place for now, or only defined it if compiled with Elixir 1.17+.

Comment (whatyouhide) [reply lib/google/protobuf.ex:18]: Same for `from_time_unit`

File: lib/google/protobuf.ex [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+
+  def from_time_unit(millisecond, :nanosecond) do
+    struct(Google.Protobuf.Duration, %{
+      seconds: div(millisecond, 1_000_000_000),
+      nanos: rem(millisecond, 1_000_000_000)
+    })
+  end
+
+  @doc """
+  Converts a `Google.Protobuf.Struct` struct to a `map()` recursively
```
Comment (whatyouhide) [inline lib/google/protobuf.ex:68]: Please use the correct ExDoc syntax for linking types:

```suggestion
  Converts a `Google.Protobuf.Struct` struct to a `t:map/0` recursively
```

**Comment by v0idpwn** [comment]
@btkostner, would you like to address Andrea's comments, or would you prefer for me to take this one over? Happy either way :)

**Comment by btkostner** [comment]
Go for it! I'm on a deadline for work right now so I doubt I'll have free time to work on this 👍

File: lib/google/protobuf.ex (hunk 1 above)
Comment (v0idpwn) [reply lib/google/protobuf.ex]: Removed them for now, we can consider adding them at some point :)

**Comment by yordis** [comment]
@v0idpwn do you think it would be possible to cut off a release?

**Comment by v0idpwn** [comment]
Hi, @yordis! I don't have the permission to make releases, so we need to ask @whatyouhide or @ericmj. 

For me, the current main seems okay to be released. I'm just a bit worried that #384 and #353 may make update not very smooth.

**Comment by yordis** [comment]
Removing the google_protos pkg will fix #353 since I just did it. Thus far, things are working for me; what is your concern with #384 per se?

**Pull request merged**